src/mongoose_rabbit_worker.erl

```diff
@@ -132,18 +135,36 @@ publish_groupchat_msg_recv(Channel, EventData = #{to_jid := To,
 publish_chat_msg_event(Channel, RoutingKey, #{from_jid := From,
                                               to_jid := To,
                                               msg := UserMsg,
-                                              exchange := Exchange}) ->
+                                              exchange := Exchange,
+                                              host := Host}) ->
     Message = make_chat_msg(From, To, UserMsg),
-    publish(Channel, Exchange, RoutingKey, Message).
+    publish(Channel, Exchange, RoutingKey, Message, Host).
 
 -spec publish(Channel :: pid(), Exchange :: binary(), RoutingKey :: binary(),
-              Message :: binary()) -> term().
-publish(Channel, Exchange, RoutingKey, Message) ->
-    amqp_channel:call(Channel,
-                      #'basic.publish'{
-                         exchange = Exchange,
-                         routing_key = RoutingKey},
-                      #amqp_msg{payload = Message}).
+              Message :: binary(), Host :: jid:server()) -> term().
+publish(Channel, Exchange, RoutingKey, Message, Host) ->
+    {PublishTime, _} = timer:tc(amqp_channel, call,
+                                [Channel,
+                                 #'basic.publish'{exchange = Exchange,
+                                                  routing_key = RoutingKey},
+                                 #amqp_msg{payload = Message}]),
+    case amqp_channel:wait_for_confirms(Channel) of
+        true ->
+            mongoose_metrics:update(Host, ?MESSAGES_PUBLISH_METRIC, 1),
+            mongoose_metrics:update(Host, ?MESSAGE_PUBLISH_TIME_METRIC,
```

Comment:
Hmm this seem to be wrong for me as you put the time it takes to actually send the message not the time span between sending it and getting the confirmation.

Reply:
It depends how we look at this. Do we want to measure time it takes to only publish a message or to publish and get the confirmation? But I think you're right and we should use the latter approach.